### PR TITLE
Set family level due date when HUB call fails 

### PR DESCRIPTION
**Pull request by hKalesnikavaideacrew**
# PR Checklist

Please check if your PR fulfills the following requirements
- [x] The title follows our [guidelines](https://github.com/ideacrew/enroll/blob/trunk/CONTRIBUTING.md#commit)
- [x] Tests for the changes have been added (for bugfixes/features), they use let helpers and before blocks
- [ ] For all UI changes, there is cucumber coverage
- [ ] Any endpoint touched in the PR has an appropriate Pundit policy. For open endpoints, reasoning is documented in PR and code
- [ ] Any endpoint modified in the PR only responds to the expected MIME types.
- [ ] For all scripts or rake tasks, how to run it is documented on both the PR and in the code
- [x] There are no inline styles added
- [x] There are no inline javascript added
- [x] There is no hard coded text added/updated in helpers/views/Javascript. New/updated translation strings do not include markup/styles, unless there is supporting documentation
- [x] Code does not use .html_safe
- [ ] All images added/updated have alt text
- [x] Doesn’t bypass rubocop rules in any way

# PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature (requires Feature flag)
- [ ] Data fix or migration (inert code, no impact until run)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Dependency updates (e.g., add a new gem or update to a version)

# What is the ticket # detailing the issue?

Ticket: https://www.pivotaltracker.com/n/projects/2640060/stories/187467013

Set family level due date when HUB call fails 

# Feature Flag

For all new feature development, a feature flag is required to control the exposure of the feature to our end users. A feature flag needs a corresponding environment variable to initialize the state of the flag. Please share the name of the environment variable below that would enable/disable the feature and which client(s) it applies to.

Variable name:

- [ ] DC
- [ ] ME

# Additional Context
Include any additional context that may be relevant to the peer review process.

# AppScan CodeSweep Failure
In the event of a failed check on the AppScan CodeSweep step of our GitHub Actions workflow, please review the False Positive protocol outlined here: appscan_codesweep/CODESWEEP_FALSE_POSITIVES_README.MD

Add all required notes to this section if the failure is a suspected false positive.

### CODE REVIEW

File: app/models/lawful_presence_determination.rb [hunk 1]
```diff
@@ -84,6 +84,7 @@ def start_ssa_process
 
       if result.failure? && EnrollRegistry.feature_enabled?(:validate_and_record_publish_errors)
         process_ssa_request_failure(result, ssa_verification_type)
+        ::Operations::Eligibilities::BuildFamilyDetermination.new.call(family: self.ivl_role.person.primary_family, effective_date: TimeKeeper.date_of_record)
```
Comment (vkghub) [inline app/models/lawful_presence_determination.rb:87]: This request can fail for multiple reason. But as per current ticket requirement if the failure is for **no VLP documents** error then reset the due date.

File: app/controllers/documents_controller.rb [hunk 2]
```diff
@@ -145,12 +145,12 @@ def enrollment_verification
...
     authorize HbxProfile, :can_call_hub?
 
-    request_hash = {person_id: @person.id, verification_type: @verification_type.type_name}
+    request_hash = { person_id: @person.id, verification_type: @verification_type.type_name }
     result = ::Operations::CallFedHub.new.call(request_hash)
     key, message = result.failure? ? result.failure : result.success
-
     if result.failure?
       @verification_type.fail_type
+      ::Operations::Eligibilities::BuildFamilyDetermination.new.call(family: @person.primary_family, effective_date: TimeKeeper.date_of_record)
```
Comment (vkghub) [inline app/controllers/documents_controller.rb:153]: This request can fail for multiple reason. But as per current ticket requirement if the failure is for no VLP documents error then reset the due date.

File: spec/controllers/documents_controller_spec.rb [hunk 3]
```diff
@@ -189,6 +189,104 @@
       end
     end
 
+    context 'enrolled' do
+      let(:application) do
```
Comment (vkghub) [inline spec/controllers/documents_controller_spec.rb:193]: I think loading applicant and application is not required for verification types. Can you remove this?

**Review by vkghub — APPROVED** [review]
- Resets the family level due date and eligibility state when the hub request fails in enroll.

LGTM

**Review by jayreddy519 — APPROVED** [review]

**Pull request merged**